File: task012_ros2/_1.创建节点/ros2_node.cpp
#include <iostream>
#include <rclcpp/rclcpp.hpp>

int main(int argc, char* argv[]) {
  rclcpp::init(argc,argv);    // 初始化ROS2,分配资源
  auto node = std::make_shared<rclcpp::Node>("cpp_node");  // 创建节点
  RCLCPP_INFO(node->get_logger(),"你好,C++节点!");  // 打印日志
  rclcpp::spin(node);  // 运行节点

  rclcpp::shutdown();  // 关闭节点
  return 0;
}

File: task012_ros2/_1.创建节点/ros2_node.py
import rclpy
from rclpy.node import Node

def main():
    rclpy.init()  # 初始化公作,分配资源
    node = Node('python_node')  # 创建节点
    node.get_logger().info('你好,python 节点!') # 打印info日志
    node.get_logger().warn('你好,python 节点!') # 打印warn日志
    node.get_logger().error('你好,python 节点!') # 打印error日志
    rclpy.spin(node) # 运行节点
    rclpy.shutdown() # 关闭节点

if __name__ == '__main__':
    main()